src/ui/verificationSymbolStatusView.ts

```diff
@@ -0,0 +1,318 @@
+      map((element, index) => {
+        return { verifiable: element, testItem: items[index] };
+      }).
+      filter(({ verifiable, testItem }) => {
+        return this.itemStates.get(testItem.id) !== verifiable.status && this.itemRuns.get(testItem.id) === undefined;
+      }).map(r => r.testItem);
+    if(runningItemsWithoutRun.length > 0) {
+      this.createRun(runningItemsWithoutRun);
+    }
+
+    const newItemRuns = new Map();
+    params.namedVerifiables.forEach((element, index) => {
+      const testItem = items[index];
+      const itemRunState = this.itemRuns.get(testItem.id)!;
+      if(this.itemStates.get(testItem.id) === element.status) {
+        newItemRuns.set(testItem.id, itemRunState);
+        return;
+      }
+      const { run, startedRunningTime } = itemRunState;
+
+      const endRun = () => {
+        const remaining = this.runItemsLeft.get(run)! - 1;
+        this.itemRuns.delete(testItem.id);
+        if(remaining === 0) {
+          this.runItemsLeft.delete(run);
+          this.automaticRunEnd = true;
+          run.end();
+          this.automaticRunEnd = false;
+        } else {
+          this.runItemsLeft.set(run, remaining);
+        }
+      };
+      const getDuration = () => startedRunningTime === undefined ? undefined : Date.now() - startedRunningTime;
+      switch(element.status) {
+      case PublishedVerificationStatus.Stale: {
+        run.skipped(testItem);
+        endRun();
+        break;
+      }
+      case PublishedVerificationStatus.Error:
+        run.failed(testItem, [], getDuration());
+        endRun();
+        break;
+      case PublishedVerificationStatus.Correct:
+        run.passed(testItem, getDuration());
+        endRun();
+        break;
+      case PublishedVerificationStatus.Running:
+        run.started(testItem);
+        newItemRuns.set(testItem.id, { run, startedRunningTime: Date.now() });
+        break;
+      case PublishedVerificationStatus.Queued:
+        run.enqueued(testItem);
+        newItemRuns.set(testItem.id, itemRunState);
+        break;
+      }
+    });
+    this.itemStates = new Map(params.namedVerifiables.map((v, index) => [ items[index].id, v.status ]));
+    this.itemRuns = newItemRuns;
+  }
+
+  private async updateStatusBar(params: IVerificationSymbolStatusParams) {
+    const completed = params.namedVerifiables.filter(v => v.status >= PublishedVerificationStatus.Error).length;
+    const running = params.namedVerifiables.filter(v => v.status === PublishedVerificationStatus.Running);
+    const total = params.namedVerifiables.length;
+    let message: string;
+    if(running.length > 0) {
+      let verifying: string;
+      if(running.length === 1) {
+        const document = await workspace.openTextDocument(Uri.parse(params.uri));
+        verifying = document.getText(VerificationSymbolStatusView.convertRange(running[0].nameRange));
+      } else {
+        verifying = `${running.length} proofs`;
```

Comment:
"proof" is a bit vague because there can be multiple proofs in a single method. I suggest we say we verified "declarations" or "definitions", which I think is more accurate.

```suggestion
        verifying = `${running.length} top-level declarations`;
```
or
```suggestion
        verifying = `${running.length} definitions`;
```
or
```suggestion
        verifying = `${running.length} declarations`;
```